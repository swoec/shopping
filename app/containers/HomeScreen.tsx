import React, { Component } from "react";
import { View, Text, StyleSheet, Button } from "react-native";

class HomeScreen extends Component {
  public props: any;
  render() {
    return (
      <View style={styles.container}>
        <Button
          title="start"
          onPress={() => this.props.navigation.navigate("Books")}
        />
      </View>
    );
  }
}
export default HomeScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
});
