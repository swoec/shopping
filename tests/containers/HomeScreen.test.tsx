import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { View, Button } from "react-native";
import HomeScreen from "../../app/containers/HomeScreen";
import {
  NavigationParams,
  NavigationScreenProp,
  NavigationState,
  withNavigation,
} from "react-navigation";

const createTestProps = (props: any) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe("Test Home Screen", () => {
  let wrapper: ShallowWrapper;
  let props: any;
  beforeEach(() => {
    props = createTestProps({});
    wrapper = shallow(<HomeScreen {...props} />);
  });

  it("It should match initial snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("It should call navigation when button is clicked", () => {
    wrapper
      .find(Button)
      .props()
      .onPress({} as any);
    expect(props.navigation.navigate).toHaveBeenCalledWith("Books");
  });
});
