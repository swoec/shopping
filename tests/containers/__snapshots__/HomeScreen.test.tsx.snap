// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Screen It should match initial snapshot 1`] = `ShallowWrapper {}`;
